<a href="https://colab.research.google.com/github/Meenakshipooranik/AI-Financial-ChatBot/blob/main/Financial_Data.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [6]:
import pandas as pd

In [7]:
df=pd.read_csv('/content/drive/MyDrive/BCG/10k.csv')

In [8]:
df

Company  Year    Total Revenue      Net Income     Total Assets  \
0  Microsoft  2023  211,915,000,000  72,361,000,000  411,976,000,000   
1  Microsoft  2022  198,270,000,000  72,738,000,000  364,840,000,000   
2  Microsoft  2021  168,088,000,000  61,271,000,000  333,779,000,000   
3      Tesla  2023   96,773,000,000  14,974,000,000  106,618,000,000   
4      Tesla  2022   81,462,000,000  12,587,000,000   82,338,000,000   
5      Tesla  2021   53,823,000,000   5,644,000,000   62,131,000,000   
6      Apple  2023  383,285,000,000  96,995,000,000  352,583,000,000   
7      Apple  2022  394,328,000,000  99,803,000,000  352,755,000,000   
8      Apple  2021  365,817,000,000  94,680,000,000  351,022,000,000   

  Total Liabilities        Cash Flow  
0   205,753,000,000   87,582,000,000  
1   198,298,000,000   89,035,000,000  
2   191,791,000,000   76,740,000,000  
3    43,009,000,000   13,256,000,000  
4    36,440,000,000   14,724,000,000  
5    30,548,000,000   11,497,000,000  
6   290,437,000,000  110,543,000,000  
7   302,083,000,000  122,151,000,000  
8   287,912,000,000  104,038,000,000

# Analyzing trends with pandas:


In [9]:
def remove_commas(value):
    value = value.split(',')
    value = ''.join(value)
    return int(value)
for col in df.columns:
    if col not in ["Company", "Year"]:
        df[col] = df[col].map(remove_commas)

In [10]:
df

Company  Year  Total Revenue   Net Income  Total Assets  \
0  Microsoft  2023   211915000000  72361000000  411976000000   
1  Microsoft  2022   198270000000  72738000000  364840000000   
2  Microsoft  2021   168088000000  61271000000  333779000000   
3      Tesla  2023    96773000000  14974000000  106618000000   
4      Tesla  2022    81462000000  12587000000   82338000000   
5      Tesla  2021    53823000000   5644000000   62131000000   
6      Apple  2023   383285000000  96995000000  352583000000   
7      Apple  2022   394328000000  99803000000  352755000000   
8      Apple  2021   365817000000  94680000000  351022000000   

   Total Liabilities     Cash Flow  
0       205753000000   87582000000  
1       198298000000   89035000000  
2       191791000000   76740000000  
3        43009000000   13256000000  
4        36440000000   14724000000  
5        30548000000   11497000000  
6       290437000000  110543000000  
7       302083000000  122151000000  
8       287912000000  104038000000

In [11]:
df = df.sort_values(['Company','Year'],ascending=True).groupby('Company').head()
df

Company  Year  Total Revenue   Net Income  Total Assets  \
8      Apple  2021   365817000000  94680000000  351022000000   
7      Apple  2022   394328000000  99803000000  352755000000   
6      Apple  2023   383285000000  96995000000  352583000000   
2  Microsoft  2021   168088000000  61271000000  333779000000   
1  Microsoft  2022   198270000000  72738000000  364840000000   
0  Microsoft  2023   211915000000  72361000000  411976000000   
5      Tesla  2021    53823000000   5644000000   62131000000   
4      Tesla  2022    81462000000  12587000000   82338000000   
3      Tesla  2023    96773000000  14974000000  106618000000   

   Total Liabilities     Cash Flow  
8       287912000000  104038000000  
7       302083000000  122151000000  
6       290437000000  110543000000  
2       191791000000   76740000000  
1       198298000000   89035000000  
0       205753000000   87582000000  
5        30548000000   11497000000  
4        36440000000   14724000000  
3        43009000000   13256000000

In [12]:
df['Revenue Growth (%)'] = df.groupby(['Company'])['Total Revenue'].pct_change() * 100
df['Net Income Growth (%)'] = df.groupby(['Company'])['Net Income'].pct_change() * 100

In [13]:
df

Company  Year  Total Revenue   Net Income  Total Assets  \
8      Apple  2021   365817000000  94680000000  351022000000   
7      Apple  2022   394328000000  99803000000  352755000000   
6      Apple  2023   383285000000  96995000000  352583000000   
2  Microsoft  2021   168088000000  61271000000  333779000000   
1  Microsoft  2022   198270000000  72738000000  364840000000   
0  Microsoft  2023   211915000000  72361000000  411976000000   
5      Tesla  2021    53823000000   5644000000   62131000000   
4      Tesla  2022    81462000000  12587000000   82338000000   
3      Tesla  2023    96773000000  14974000000  106618000000   

   Total Liabilities     Cash Flow  Revenue Growth (%)  Net Income Growth (%)  
8       287912000000  104038000000                 NaN                    NaN  
7       302083000000  122151000000            7.793788               5.410858  
6       290437000000  110543000000           -2.800461              -2.813543  
2       191791000000   76740000000                 NaN                    NaN  
1       198298000000   89035000000           17.956071              18.715216  
0       205753000000   87582000000            6.882030              -0.518299  
5        30548000000   11497000000                 NaN                    NaN  
4        36440000000   14724000000           51.351653             123.015592  
3        43009000000   13256000000           18.795267              18.964010

In [14]:
df.fillna(0, inplace=True)
df

Company  Year  Total Revenue   Net Income  Total Assets  \
8      Apple  2021   365817000000  94680000000  351022000000   
7      Apple  2022   394328000000  99803000000  352755000000   
6      Apple  2023   383285000000  96995000000  352583000000   
2  Microsoft  2021   168088000000  61271000000  333779000000   
1  Microsoft  2022   198270000000  72738000000  364840000000   
0  Microsoft  2023   211915000000  72361000000  411976000000   
5      Tesla  2021    53823000000   5644000000   62131000000   
4      Tesla  2022    81462000000  12587000000   82338000000   
3      Tesla  2023    96773000000  14974000000  106618000000   

   Total Liabilities     Cash Flow  Revenue Growth (%)  Net Income Growth (%)  
8       287912000000  104038000000            0.000000               0.000000  
7       302083000000  122151000000            7.793788               5.410858  
6       290437000000  110543000000           -2.800461              -2.813543  
2       191791000000   76740000000            0.000000               0.000000  
1       198298000000   89035000000           17.956071              18.715216  
0       205753000000   87582000000            6.882030              -0.518299  
5        30548000000   11497000000            0.000000               0.000000  
4        36440000000   14724000000           51.351653             123.015592  
3        43009000000   13256000000           18.795267              18.964010

In [15]:
df['Assets Growth (%)'] = df.groupby(['Company'])['Total Assets'].pct_change() * 100
df['Liabilities Growth (%)'] = df.groupby(['Company'])['Total Liabilities'].pct_change() * 100
df['Cashflow Growth (%)'] = df.groupby(['Company'])['Cash Flow'].pct_change() * 100
df

Company  Year  Total Revenue   Net Income  Total Assets  \
8      Apple  2021   365817000000  94680000000  351022000000   
7      Apple  2022   394328000000  99803000000  352755000000   
6      Apple  2023   383285000000  96995000000  352583000000   
2  Microsoft  2021   168088000000  61271000000  333779000000   
1  Microsoft  2022   198270000000  72738000000  364840000000   
0  Microsoft  2023   211915000000  72361000000  411976000000   
5      Tesla  2021    53823000000   5644000000   62131000000   
4      Tesla  2022    81462000000  12587000000   82338000000   
3      Tesla  2023    96773000000  14974000000  106618000000   

   Total Liabilities     Cash Flow  Revenue Growth (%)  Net Income Growth (%)  \
8       287912000000  104038000000            0.000000               0.000000   
7       302083000000  122151000000            7.793788               5.410858   
6       290437000000  110543000000           -2.800461              -2.813543   
2       191791000000   76740000000            0.000000               0.000000   
1       198298000000   89035000000           17.956071              18.715216   
0       205753000000   87582000000            6.882030              -0.518299   
5        30548000000   11497000000            0.000000               0.000000   
4        36440000000   14724000000           51.351653             123.015592   
3        43009000000   13256000000           18.795267              18.964010   

   Assets Growth (%)  Liabilities Growth (%)  Cashflow Growth (%)  
8                NaN                     NaN                  NaN  
7           0.493701                4.921990            17.409985  
6          -0.048759               -3.855232            -9.502992  
2                NaN                     NaN                  NaN  
1           9.305858                3.392756            16.021631  
0          12.919636                3.759493            -1.631942  
5                NaN                     NaN                  NaN  
4          32.523217               19.287678            28.068192  
3          29.488207               18.026894            -9.970117

In [16]:
df.fillna(0, inplace=True)
df

Company  Year  Total Revenue   Net Income  Total Assets  \
8      Apple  2021   365817000000  94680000000  351022000000   
7      Apple  2022   394328000000  99803000000  352755000000   
6      Apple  2023   383285000000  96995000000  352583000000   
2  Microsoft  2021   168088000000  61271000000  333779000000   
1  Microsoft  2022   198270000000  72738000000  364840000000   
0  Microsoft  2023   211915000000  72361000000  411976000000   
5      Tesla  2021    53823000000   5644000000   62131000000   
4      Tesla  2022    81462000000  12587000000   82338000000   
3      Tesla  2023    96773000000  14974000000  106618000000   

   Total Liabilities     Cash Flow  Revenue Growth (%)  Net Income Growth (%)  \
8       287912000000  104038000000            0.000000               0.000000   
7       302083000000  122151000000            7.793788               5.410858   
6       290437000000  110543000000           -2.800461              -2.813543   
2       191791000000   76740000000            0.000000               0.000000   
1       198298000000   89035000000           17.956071              18.715216   
0       205753000000   87582000000            6.882030              -0.518299   
5        30548000000   11497000000            0.000000               0.000000   
4        36440000000   14724000000           51.351653             123.015592   
3        43009000000   13256000000           18.795267              18.964010   

   Assets Growth (%)  Liabilities Growth (%)  Cashflow Growth (%)  
8           0.000000                0.000000             0.000000  
7           0.493701                4.921990            17.409985  
6          -0.048759               -3.855232            -9.502992  
2           0.000000                0.000000             0.000000  
1           9.305858                3.392756            16.021631  
0          12.919636                3.759493            -1.631942  
5           0.000000                0.000000             0.000000  
4          32.523217               19.287678            28.068192  
3          29.488207               18.026894            -9.970117

In [17]:
df.to_csv('/content/drive/MyDrive/BCG/Analysis.csv')

In [18]:
summary = df.groupby('Company').agg({
    'Revenue Growth (%)': 'mean',
    'Net Income Growth (%)': 'mean',
    'Assets Growth (%)': 'mean',
    'Liabilities Growth (%)': 'mean',
    'Cashflow Growth (%)': 'mean'
}).reset_index()

In [19]:
summary

Company  Revenue Growth (%)  Net Income Growth (%)  Assets Growth (%)  \
0      Apple            1.664442               0.865772           0.148314   
1  Microsoft            8.279367               6.065639           7.408498   
2      Tesla           23.382306              47.326534          20.670475   

   Liabilities Growth (%)  Cashflow Growth (%)  
0                0.355586             2.635664  
1                2.384083             4.796563  
2               12.438191             6.032692

In [20]:
summary.to_csv('/content/drive/MyDrive/BCG/Summary_Report_Data.csv')


**TRENDS**:

Summary of Financial Analysis Findings over 3 year fiscal period

**Revenue Growth Trends:**

Apple showed a modest average revenue growth of 1.664442%.
Microsoft had a higher average revenue growth rate of 8.279367%.
Tesla had a substantial average revenue growth of 23.382306%.

**Net Income Growth Trends:**

Apple's net income growth averaged 0.865772%.
Microsoft and Tesla also showed varying net income growth rates, 6.0656639% and 48.920903% respectively, indicating differing levels of profitability and cost management efficiency.

**Assets and Liabilities Growth:**

Apple's assets grew on average by 0.150223%, while liabilities growth was minimal at 0.355586%, suggesting stable financial management.
Microsoft and Tesla revealed more significant fluctuations in asset and liability growth rates, reflecting different investment or financial strategies.

**Cash Flow from Operating Activities:**

Growth rates varied significantly among companies, with some showing strong growth in operational cash flow, indicating robust operational health.
Analysis Narratives

**Revenue and Net Income Growth:**

The analysis indicates a general trend of growth in revenue across the companies, with Microsoft showing a particularly strong increase. This suggests effective market penetration and product acceptance.
Net income growth rates are less consistent, which could reflect varying cost structures and operational efficiencies. Apple's relatively stable net income growth suggests effective cost management despite modest revenue growth.

**Asset Management:**

The asset growth rates are generally low, suggesting cautious capital management. Companies with minimal growth in liabilities alongside asset growth are likely improving their financial leverage and stability.

**Operational Efficiency:**

Variations in cash flow from operating activities growth rates highlight differences in operational efficiency and cash management strategies among the companies. High growth in this area typically indicates more efficient operations and better cash management.

# **SUMMARY:**
Tesla has shown the highest growth in all metrics, particularly in net income and revenue, indicating significant expansion and profitability in recent years. Microsoft follows with solid growth across all categories, while Apple shows more modest growth rates but still positive gains across the board.